instrumentation/rxjava-2.0/javaagent/rxjava-2.0-javaagent.gradle

```diff
@@ -0,0 +1,17 @@
+apply from: "$rootDir/gradle/instrumentation.gradle"
+
+muzzle {
+  pass {
+    group = "io.reactivex.rxjava2"
+    module = "rxjava"
+    versions = "[2.0.0,)"
+  }
+}
+
+dependencies {
+  implementation group: 'org.reactivestreams', name: 'reactive-streams', version: '1.0.0'
+  library group: 'io.reactivex.rxjava2', name: 'rxjava', version: '2.0.0'
```

Comment:
Does this instrumentation use rxjava, it seems to only instrument reactive streams? I couldn't find any usage outside of tests.  If so we can name the instrumentation reactivestreams (we'd probably want to add tests with reactor too). Also I wonder how this instrumentation interops with other reactivestreams instrumentation we have, such as `reactor`.

Also for this instrumentation especially, I think it would help to start with library instrumentation instead of javaagent instrumentation - It uses so much bytecode instrumentation I don't think we could extract a library for it. An example library that instruments rxjava2 for context propagation is https://github.com/line/armeria/tree/master/rxjava2

Reply:
I was porting DD instrumentation as mentioned here https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1866. And the DD instrumentation uses javaagent instrumentation (https://github.com/DataDog/dd-trace-java/tree/master/dd-java-agent/instrumentation/rxjava-2)

Reply:
@piotr-sumo Would you be able to help us by extracting this into library instrumentation? Going forward we do want to make sure we have it where we can. If it's too much for now though, I think this time we can stick with this and refactor (a fair amount I guess) later.

Reply:
@anuraaga Sure, I'd like to contribute and extract this into library instrumentation. First, I'll familiarise myself with https://github.com/line/armeria/tree/master/rxjava2 and proceed with library instrumentation for `rxjava-2`.